core/src/stores/postgres.rs

```diff
@@ -1844,307 +1845,304 @@ impl Store for PostgresStore {
-                    SELECT sqlite_worker 
-                    FROM databases
-                    WHERE data_source = $1 AND database_id = $2
-                 ) AND last_heartbeat > $3 LIMIT 1",
+                "SELECT table_ids_hash \
+                FROM databases \
+                WHERE table_ids_hash LIKE $1",
             )
             .await?;
-        let r = tx
+        let r = c
             .query(
                 &stmt,
-                &[
-                    &data_source_row_id,
-                    &database_id,
-                    &((utils::now() - ttl) as i64),
-                ],
+                &[&format!(
+                    "%{}%",
+                    get_table_unique_id(project, &data_source_id, &table_id)
+                )],
             )
             .await?;
+        let database_ids = r
+            .into_iter()
+            .map(|row| row.get::<usize, String>(0))
+            .collect::<Vec<_>>();
 
-        let worker: Option<SqliteWorker> = match r.len() {
-            0 => None,
-            1 => {
-                let (url, last_heartbeat): (String, i64) = (r[0].get(0), r[0].get(1));
-                Some(SqliteWorker::new(url, last_heartbeat as u64))
-            }
-            _ => unreachable!(),
-        };
-
-        if worker.is_some() {
-            // There is already an assigned worker.
-            // We can release the lock and return the worker.
-            tx.commit().await?;
-            return Ok(worker.unwrap());
-        }
-
-        // Pick a random live worker.
-        let stmt = tx
-            .prepare(
-                "SELECT id, url, last_heartbeat 
-                FROM sqlite_workers 
-                WHERE last_heartbeat > $1 ORDER BY RANDOM() LIMIT 1",
-            )
-            .await?;
-        let r = tx.query(&stmt, &[&((utils::now() - ttl) as i64)]).await?;
-
-        match r.len() {
-            0 => Err(anyhow!("No live workers found"))?,
-            1 => {
-                let (sqlite_worker_row_id, url, last_heartbeat): (i64, String, i64) =
-                    (r[0].get(0), r[0].get(1), r[0].get(2));
-
-                // Update the database row to assign the worker.
-                let stmt = tx
-                    .prepare(
-                        "UPDATE databases SET sqlite_worker = $1 \
-                        WHERE data_source = $2 AND database_id = $3",
-                    )
-                    .await?;
-                tx.execute(
-                    &stmt,
-                    &[
-                        &sqlite_worker_row_id,
-                        &data_source_row_id,
-                        &database_id.to_string(),
-                    ],
-                )
-                .await?;
-
-                // Release the lock.
-                tx.commit().await?;
-
-                Ok(SqliteWorker::new(url, last_heartbeat as u64))
-            }
-            _ => unreachable!(),
-        }
+        Ok((try_join_all(
```

Comment:
This is not perfect, but a one table would probably not be used by that many tables. We have N+1 queries here (as we need to get the sqlite worker for each db). If we don't want this, it will result in much more complex code.

IMO it's a fair tradeoff here cc @spolu